CLAUDE.md

```diff
@@ -0,0 +1,276 @@
+`scripts/variantstore/docs/Build Docker from VM/building_gatk_docker_on_a_vm.md`
+file. An x86 cloud-based VM is used to build this x86 Docker image, which is
+then pushed to the Broad's GAR (Google Artifact Registry) repository.
+
+### Variants Docker Image
+
+This is a Google Cloud SDK Alpine-based Docker image to which many useful tools,
+Python packages, and scripts have been added. This can be built on a developer's
+local machine using the `scripts/variantstore/scripts/build_docker.sh` script.
+
+### Nirvana Docker Image
+
+This is an Illumina Nirvana-based Docker image build from
+`scripts/variantstore/variant-annotations-table/build_docker.sh`. Since Nirvana
+is no longer being updated this image hopefully will also not need to be updated
+in the future.
+
+## "location" versus "position"
+
+GVS leverages BigQuery partitioning and clustering to optimize query performance
+and cost. The core tables are either partitioned by sample ID or location,
+depending on the access pattern for which they are optimized. GVS uses the term
+"location" in a very specific way: the chromosome number is multiplied by
+1,000,000,000,000 and then the position on the chromosome is added. For example,
+chromosome 1 position 1000 becomes `1000000000000 + 1000 = 1000000001000`. GVS
+locations use a chromosome number of 23 for the X chromosome and 24 for the Y
+chromosome. This location encoding allows for efficient storage and querying in
+BigQuery using a single partitionable value. GVS references to "position"
+usually refer to position on a given chromosome and do not mean the same thing
+as "location".
+
+## Core BigQuery Tables
+
+- **`sample_info`**: Holds sample name, control / non-control status, and a
+  withdrawal date if applicable.
+- **`sample_load_status`**: This table tracks which data has been loaded for
+  each sample. As described in the LoadStatusValue enum in LoadStatus.java, the
+  values currently used include REFERENCES_LOADED, VARIANTS_LOADED,
+  HEADERS_LOADED. For historical reasons some legacy values are also recognized.
+- **`sample_chromosome_ploidy`**: This table contains a per-sample,
+  per-chromosome of the ploidy observed in its input GVCF file.
+- **`vet_%`**: Primary variant information (sample id, location, alleles,
+  annotations). These tables are partitioned by sample id. Due to BigQuery's
+  4000 partition limit there can be many `vet_%` tables in a GVS dataset, named
+  `vet_001`, `vet_002`, etc. Schema information for vet tables can be found in
+  the `CreateBQTables.wdl` workflow in the variable `vet_schema_json`.
+- **`ref_ranges_%`**: Reference genome locations, lengths, and qualities.
+  Similar to vet, this table is partitioned by sample id and can hold at most
+  4000 samples, so there will be multiple `ref_ranges_%` for larger datasets.
+  Note that `ref_ranges_%` tables have two alternate schemas, compressed and
+  uncompressed, to accommodate different data storage needs. Both schemas are
```

Comment:
We'll likely want to point it to the code here that encodes all of the data into a single Long (and decodes that as well) in SchemaUtils.encodeCompressedRefBlock